### PR TITLE
[MM-20572] Migrate 'components/admin_console/compliance_reports' to TypeScript

### PR DESCRIPTION
**Pull request by daniel-shuy**
#### Summary

Migrate all files in `components/admin_console/compliance_reports` to TypeScript.

#### Ticket Link

Closes mattermost/mattermost-server#13500

### CODE REVIEW

File: components/admin_console/compliance_reports/compliance_reports.tsx [hunk 1]
```diff
@@ -1,57 +1,70 @@
...
-            /*
-             * Function to save compliance reports
-             */
-            createComplianceReport: PropTypes.func.isRequired,
-        }).isRequired,
-    }
+        /*
+            * Function to save compliance reports
+            */
+        createComplianceReport: (job: Partial<Compliance>) => Promise<{data: Compliance, error?: Error}>;
```
Comment (daniel-shuy) [inline components/admin_console/compliance_reports/compliance_reports.tsx:51]: I'm not sure if this is a bug in `mattermost-webapp` or `mattermost-redux`, but the [createComplianceReport()](https://github.com/mattermost/mattermost-redux/blob/5aa938486c49d76f55953dca7890fb6f4ad601b7/src/client/client4.ts#L2656) function in `mattermost-redux` takes in a `Job`, however, if you look at the type definition of [Job](https://github.com/mattermost/mattermost-redux/blob/5aa938486c49d76f55953dca7890fb6f4ad601b7/src/types/jobs.ts#L6), it does not match the usage in [runReport()](https://github.com/mattermost/mattermost-webapp/pull/7036/files#diff-404f3aee8367f0adaedc23366b864a9ce42e47b0fea4cec46fa15421e36f403dR104-R109) below, which is why I used `Partial<Compliance>` instead.

Comment (daniel-shuy) [reply components/admin_console/compliance_reports/compliance_reports.tsx:51]: Looks like this is a bug in `mattermost-redux`. The [`/compliance/reports`](https://github.com/mattermost/mattermost-server/blob/v5.28.1/api4/compliance.go#L23) API in `mattermost-server` is expecting a [Compliance](https://github.com/mattermost/mattermost-redux/blob/5aa938486c49d76f55953dca7890fb6f4ad601b7/src/types/compliance.ts), not a [Job](https://github.com/mattermost/mattermost-redux/blob/5aa938486c49d76f55953dca7890fb6f4ad601b7/src/types/jobs.ts#L6), see:
- https://github.com/mattermost/mattermost-server/blob/v5.28.1/model/compliance.go#L101
- https://github.com/mattermost/mattermost-server/blob/v5.28.1/model/compliance.go#L24

Comment (daniel-shuy) [reply components/admin_console/compliance_reports/compliance_reports.tsx:51]: Created a separate PR for this at mattermost/mattermost-redux#1289

**Review by hahmadia — APPROVED** [review]
Great work and nice catch on the bug! Thanks!

**Review by mkraft — APPROVED** [review]

**Review by jgilliam17 — APPROVED** [review]
Thank you @daniel-shuy
Tested, looks good to merge.
- Verified compliance report section in system console.

**Pull request merged**